py-polars/polars/dataframe/frame.py

```diff
@@ -2286,7 +2283,40 @@ def to_pandas(  # noqa: D417
                 else:
                     raise ModuleNotFoundError(msg)
 
-        record_batches = self._df.to_pandas()
+        object_columns = [
+            name for (name, dtype) in zip(self.columns, self.dtypes) if dtype == Object
+        ]
+        object_columns_set = set(object_columns)
```

Comment:
Can we add a fast path for the cases where we don't have any object columns?

As it is now we always go into the object related branches.